src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -164,7 +164,7 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                     diagnostics.Add(ErrorCode.ERR_ConstantExpected, patternExpression.Location);
                     hasErrors = true;
                 }
-                else if (inputType.IsPointerType() == true && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
+                else if (inputType.IsPointerOrFunctionPointer() == true && Compilation.LanguageVersion < MessageID.IDS_FeatureRecursivePatterns.RequiredVersion())
```

Comment:
>inputType.IsPointerOrFunctionPointer() [](start = 25, length = 38)

This feels suspicious. Why would we care about this language version check for function pointers, when there is no prior behavior to enforce? Also, if we were to get into the ```if```, we would complain about language version in relation to function pointers elsewhere.

Reply:
It was purely for error message consistency, but I can revert if you prefer.

---
In reply to: [407717792](https://github.com/dotnet/roslyn/pull/43248#discussion_r407717792) [](ancestors = 407717792)

Reply:
> It was purely for error message consistency, but I can revert if you prefer.

I think the experience is better without this noise around language version.

---
In reply to: [407743217](https://github.com/dotnet/roslyn/pull/43248#discussion_r407743217) [](ancestors = 407743217,407717792)